packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -436,18 +442,28 @@ export function getValidatorApi({
 
       const fork = config.getForkName(slot);
       // set some sensible opts
+      // builderSelection will be deprecated and will run in mode MaxProfit if builder is enabled
+      // and the actual selection will be determined using builderBoostFactor passed by the validator
       builderSelection = builderSelection ?? routes.validator.BuilderSelection.MaxProfit;
+      builderBoostFactor = builderBoostFactor ?? BigInt(100);
+      if (builderBoostFactor > MAX_BUILDER_BOOST_FACTOR) {
+        throw Error(`Invalid builderBoostFactor=${builderBoostFactor} > MAX_BUILDER_BOOST_FACTOR`);
+      }
+
       const isBuilderEnabled =
         ForkSeq[fork] >= ForkSeq.bellatrix &&
         chain.executionBuilder !== undefined &&
-        builderSelection !== routes.validator.BuilderSelection.ExecutionOnly;
+        builderSelection !== routes.validator.BuilderSelection.ExecutionOnly &&
+        builderBoostFactor !== BigInt(0);
```

Comment:
Does this mean if someone passed a value of 0 we won't query the builder block at all?

I think as per spec it should still query the builder in case execution block production fails
> * `builder_comparison_factor=0`: prefer the execution node payload unless an error makes it
          unviable.

Reply:
yes i thought we can skip calling builder, but yea engine block could fail making it the only viable response

Reply:
updated, thanks for noticing :+1: